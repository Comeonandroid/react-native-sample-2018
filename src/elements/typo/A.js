import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { Text } from '../index'


const A = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

A.propTypes = {
  children: PropTypes.node
}

const Element = styled(Text) `
  color: ${ ({ theme }) => theme.colors.primary }
  font-family: Open Sans;
  font-weight: 400;
`

export default A