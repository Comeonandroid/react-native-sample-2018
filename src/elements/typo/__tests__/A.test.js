import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import A from '../A'

describe('<A />', () => {
  itBehavesLike('aThemedComponent', { subject: <A /> })
  itBehavesLike('aComponentWithChildren', { subject: <A /> })
})
