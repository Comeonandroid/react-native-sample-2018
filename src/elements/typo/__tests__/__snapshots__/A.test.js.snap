// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#fcfcfc",
        "fontFamily": "Open Sans",
      },
      Array [
        Object {
          "color": "#faa632",
          "fontFamily": "Open Sans",
          "fontWeight": "400",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<A /> light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Open Sans",
      },
      Array [
        Object {
          "color": "#faa632",
          "fontFamily": "Open Sans",
          "fontWeight": "400",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<A /> renders component when passed in 1`] = `
<Styled(Text)>
  <Component />
</Styled(Text)>
`;

exports[`<A /> renders text when passed in 1`] = `
<Styled(Text)>
  Test
</Styled(Text)>
`;
